### DISCUSSION TITLE
Add Tinybird Connector

### DISCUSSION THREAD

**Discussion by ackdav** [apache/superset#21932] [Ideas]
We would like to use Superset, but would need to have an adapter to connect to [Tinybird](https://www.tinybird.co/). Tinybird uses Clickhouse under the hood, so it might be possible to re-use the existing connector.
Currently we use Redash and were able to build such a clickhouse-adapted query runner relatively quickly (see open PR for more details [here](https://github.com/getredash/redash/pull/5616)).
I was wondering if I can similarly adapt the clickhouse connection and if not, what kind process/effort it would be to build such a tinybird connector.
Thank you

**Comment by sdairs** [comment]
Hi, Alasdair from Tinybird here, happy to help support!

At the moment, we have a beta feature that can expose a PostgreSQL compatible endpoint for you to use (let us know & we can discuss that).

However, it would be awesome to be able to integrate with SuperSet over the HTTP APIs that are built in Tinybird.

I would love to get some feedback from someone with experience building connectors for SuperSet on whether this is feasible.

For context, Tinybird users write SQL inside Tinybird, and then publish the results of those SQL queries as HTTP APIs. So the natural way for us to be used in a visualisation tool is simply for the tool to hit the URL and then map the results to the visualisation.

Example:

A user writes `SELECT columnA, columnB FROM mytable` in Tinybird, and then publishes this as an API `https://api.tinybird.co/v0/pipes/my_example_pipe.json`. To run the query & get the results, you simply hit that URL, and get a JSON (or CSV/NDJSON/Parquet) response that might look like this:

```
{
  "meta": [
    {
      "name": "columnA",
      "type": "String"
    },
    {
      "name": "columnB",
      "type": "String"
    }
  ],
  "data": [
    {
      "columnA": "row1",
      "columnB": "row1"
    },
    {
      "columnA": "row2",
      "columnB": "row2"
    }
  ],
  "rows": 2,
  "rows_before_limit_at_least": 2,
  "statistics": {
    "elapsed": 0.000082874,
    "rows_read": 2,
    "bytes_read": 2
  }
}
```

So we'd want to be able to map on the rows in the `data` key, just as if they were the results of executing SQL against any other DB.

Is it feasible that we could build a connector to work like this?
